src/celeritas/em/interactor/SeltzerBergerInteractor.hh

```diff
@@ -149,17 +153,14 @@ CELER_FUNCTION Interaction SeltzerBergerInteractor::operator()(Engine& rng)
         return Interaction::from_failure();
     }
 
-    // Sample the bremsstrahlung photon energy to construct the final sampler
-    detail::BremFinalStateHelper sample_interaction(inc_energy_,
-                                                    inc_direction_,
-                                                    inc_momentum_,
-                                                    shared_.electron_mass,
-                                                    shared_.ids.gamma,
-                                                    sample_photon_energy_(rng),
-                                                    secondaries);
-
     // Update kinematics of the final state and return this interaction
-    return sample_interaction(rng);
+    return detail::BremFinalStateHelper{inc_energy_,
+                                        inc_direction_,
+                                        inc_momentum_,
+                                        shared_.ids.gamma,
+                                        sample_photon_energy_(rng),
+                                        sample_costheta_(rng),
```

Comment:
Ooh this can't be done reproducibly. There was once a hard to track discrepancy in our CI between intel/gcc because intel would sample one function first and GCC would sample the other.

Reply:
Ah, I thought because I used brace initialization here they would always be sampled in the order they appear?

Reply:
From the standard:
> 4   Within the initializer-list of a braced-init-list, the initializer-clauses, including any that result from pack
    expansions (14.5.3), are evaluated in the order in which they appear. That is, every value computation and
    side effect associated with a given initializer-clause is sequenced before every value computation and side
    effect associated with any initializer-clause that follows it in the comma-separated list of the initializer-list.
    [ Note: This evaluation ordering holds regardless of the semantics of the initialization; for example, it applies
    when the elements of the initializer-list are interpreted as arguments of a constructor call, even though
    ordinarily there are no sequencing constraints on the arguments of a call. -- end note ]

So from the note it sounds like it should be true (I initially did make this mistake, but caught it when the SB and RB tests started failing).

Reply:
Wow, that's a deep cut. Good find! Hopefully the compilers all follow this standard 😅